azure/scope/identity.go

```diff
@@ -161,3 +163,45 @@ func getAzureIdentityType(identity *infrav1.AzureClusterIdentity) (aadpodv1.Iden
 	return 0, errors.New("AzureIdentity does not have a vaild type")
 
 }
+
+// IsClusterNamespaceAllowed indicates if the cluster namespace is allowed
+func IsClusterNamespaceAllowed(ctx context.Context, k8sClient client.Client, allowedNamespaces *infrav1.AllowedNamespaces, namespace string) bool {
```

Comment:
@devigned can you take a look at this and see if it makes sense based on all the previous conversations?